File: Check_whether_BST_contains_Dead_End.py
class Solution:
    def has_dead_end(self, root, min_val, max_val):
        if root is None:
            return False
        if min_val == max_val:
            return True
        return self.has_dead_end(root.left, min_val, root.data - 1) or self.has_dead_end(root.right, root.data + 1, max_val)

    def isDeadEnd(self, root):
        return self.has_dead_end(root, 1, 10001)

File: Check_whether_BST_contains_Dead_End.java
class Solution{
    public static boolean hasDeadEnd(Node root, int minVal, int maxVal) {
        if (root == null) {
            return false;
        }
        if (minVal == maxVal) {
            return true;
        }
        return hasDeadEnd(root.left, minVal, root.data - 1) ||
            hasDeadEnd(root.right, root.data + 1, maxVal);
    }
    public static boolean isDeadEnd(Node root){
        return hasDeadEnd(root, 1, 10001);
    }
}
